build/configs/rtl8721csm/rtl8721csm_make_bin.sh

```diff
@@ -18,83 +18,100 @@
+arm-none-eabi-nm $BINDIR/target_pure_img2.axf | sort > $BINDIR/target_pure_img2.map
 
-arm-none-eabi-objcopy -Obinary /root/tizenrt/build/output/bin/target_pure_img2.axf /root/tizenrt/build/output/bin/target_img2.bin
+arm-none-eabi-objcopy -Obinary $BINDIR/target_pure_img2.axf $BINDIR/target_img2.bin
 
 arm-none-eabi-objcopy -j .ram_image2.entry -j .ram_image2.text -j .ram_image2.data \
--Obinary /root/tizenrt/build/output/bin/target_pure_img2.axf /root/tizenrt/build/output/bin/ram_2.bin
+-Obinary $BINDIR/target_pure_img2.axf $BINDIR/ram_2.bin
 
 arm-none-eabi-objcopy -j .xip_image2.text \
--Obinary /root/tizenrt/build/output/bin/target_pure_img2.axf /root/tizenrt/build/output/bin/xip_image2.bin
+-Obinary $BINDIR/target_pure_img2.axf $BINDIR/xip_image2.bin
 
 arm-none-eabi-objcopy -j .psram_image2.text -j .psram_image2.data \
--Obinary /root/tizenrt/build/output/bin/target_pure_img2.axf /root/tizenrt/build/output/bin/psram_2.bin
+-Obinary $BINDIR/target_pure_img2.axf $BINDIR/psram_2.bin
 
-arm-none-eabi-size -A --radix=16 /root/tizenrt/build/output/bin/target_img2.axf
+arm-none-eabi-size -A --radix=16 $BINDIR/target_img2.axf
 
-arm-none-eabi-size -t --radix=16 /root/tizenrt/build/output/bin/target_img2.axf
+arm-none-eabi-size -t --radix=16 $BINDIR/target_img2.axf
 
 echo "========== Image manipulating start =========="
-/root/tizenrt/build/tools/amebad/gnu_utility/prepend_header.sh /root/tizenrt/build/output/bin/ram_2.bin  __ram_image2_text_start__  /root/tizenrt/build/output/bin/target_img2.map
-/root/tizenrt/build/tools/amebad/gnu_utility/prepend_header.sh /root/tizenrt/build/output/bin/xip_image2.bin  __flash_text_start__  /root/tizenrt/build/output/bin/target_img2.map
-/root/tizenrt/build/tools/amebad/gnu_utility/prepend_header.sh /root/tizenrt/build/output/bin/psram_2.bin  __psram_image2_text_start__  /root/tizenrt/build/output/bin/target_img2.map
-python /root/tizenrt/build/tools/amebad/gnu_utility/code_analyze.py
+$GNUUTL/prepend_header.sh $BINDIR/ram_2.bin  __ram_image2_text_start__  $BINDIR/target_img2.map
+$GNUUTL/prepend_header.sh $BINDIR/xip_image2.bin  __flash_text_start__  $BINDIR/target_img2.map
+$GNUUTL/prepend_header.sh $BINDIR/psram_2.bin  __psram_image2_text_start__  $BINDIR/target_img2.map
+python $GNUUTL/code_analyze.py
 
-cat /root/tizenrt/build/output/bin/xip_image2_prepend.bin /root/tizenrt/build/output/bin/ram_2_prepend.bin /root/tizenrt/build/output/bin/psram_2_prepend.bin  > /root/tizenrt/build/output/bin/km4_image2_all.bin
+cat $BINDIR/xip_image2_prepend.bin $BINDIR/ram_2_prepend.bin $BINDIR/psram_2_prepend.bin  > $BINDIR/km4_image2_all.bin
 
-/root/tizenrt/build/tools/amebad/gnu_utility/pad.sh /root/tizenrt/build/output/bin/km4_image2_all.bin
+$GNUUTL/pad.sh $BINDIR/km4_image2_all.bin
+
+function image_encryption()
+{
+	if [ ! -f ${CONFIG} ];then
+		echo "No .config file"
+		exit 1
+	fi
+
+	source ${CONFIG}
+
+	if [ "${CONFIG_AMEBAD_TRUSTZONE}" == "y" ];then
+		RDP_KEY="${CONFIG_AMEBAD_RDP_KEY:2}"
```

Comment:
the RDP encryption depends on AMEBAD_RDP, how about check on AMEBAD_RDP rather than TRUSTZONE?

Reply:
Image encryption work like this, if trust zone is enable it will check for AMEBAD_RDP. if AMEBAD_RDP is enable it will encrypt the image by RDP_KEY and rename by adding "-en". if AMEBAD_RDP is not enable it will only rename the image by adding "-en". So at Concatenate_binary part they can use the common action without checking RDP.
Or should i modify the Concatenate_binary part?

Reply:
I got your point. Is it valid to have TRUSTZONE enabled but RDP disabled? how it ensure a secure binary if the binary is not encrypted at all?

Reply:
Yes, it is possible to have TRUSTZONE enabled but RDP disabled, we are doing this for the ease of user. Because to use RDP, user have to write the key into the board, which the RDP key can only be write once. So in this case use can test trustzone without writing RDP enable.
But for complete or final, user should enable trustzone and RDP both enable.

Reply:
ok, understood.
Can you add these comments in help section of AMEBAD_RDP config?

Reply:
Sure, i will add the comment. Thanks

Reply:
I have another question, in this case should i change the AMEBAD_RDP default to "Y" or remain as "n" becasue default user have no RDP written on board?
For  help section comment "Enable RDP Encryption for binary.  Can disbale for testing purpose due to RDP not written on board. However, should be enable for security purpose." is this okay?

Reply:
I think by default its ok to keep this disable, since it is allowed to write the key only once in a board. To make this aware to user, the option should be only enabled if user explicitly enable this.
Regarding help section, I can think of below description. But it depends upon you.
"Enable RDP Encryption for binary. RDP key once written, cannot be reprogrammed. Can disable this for testing only TrustZone without writing RDP key. However, should be enabled along with TrustZone to provide binary security. If unsure, do not enable this"

Reply:
Yours is have better explanation for the config. i will use yours as the help section. Thanks